### ISSUE TITLE
Partial hydration

### ISSUE THREAD

**Issue by zergu** [propelorm/Propel2#858]
It would be nice if we could limit columns on query with `select(array(...))` and still have collection of hydrated objects.  That could be a nice way to optimize some queries. Currently we can set object formatter for query with limited columns but hydration expects all fields and afaik in specified order which results in `Undefined offset: 1` etc. errors.

**Comment by marcj** [comment]
Yes, indeed. I think it is useful to mark all not-fetched columns as lazy-loading for the partial hydrated objects.

**Comment by MelkorNemesis** [comment]
Any update on this? At the moment I would really appreciate partial hydration because of query speed optimisation.
